src/scalarstats.jl

```diff
@@ -66,10 +88,17 @@ function mode{T<:Integer}(a::AbstractArray{T}, rgn::UnitRange{T})
     return mv
 end
 
-function modes{T<:Integer}(a::AbstractArray{T}, rgn::UnitRange{T})
-    r0 = rgn[1]
-    r1 = rgn[end]
-    n = length(rgn)
+"""
+    modes(a, [r])
+
+Return all modes (most common numbers) of an array, optionally over a
+specified range `r`. The output is always an array, even if `a` has
```

Comment:
array -> vector. But you could even remove that sentence in favor of `::Vector` in the signature. This is one example where that information is useful.